MODFQMMGR-412 Refactor how outdated queries are rejected (#801)
folio-org/mod-fqm-manager: docs/Migration.md

- `getLabel` should return a descriptive label for logging, just like `AbstractSimpleMigrationStrategy`

> - `applies` should return `true` if this migration logic should be run on a given `version`

Reviewer: ```suggestion
- `applies` can be implemented to only run migration logic on matching queries
```
